### ISSUE TITLE
ImageNet Model Requirement

### ISSUE THREAD

**Issue by chenbys** [GT-RIPL/L2C#5]
Thanks for your excellent works, but could you please share your similarity model for ImageNet?
In my experiments, the two-linear-SimilarityNet converges quickly on Omniglot, but very slowly on large scale datasets.

For example, the F1-score of "similar" for val and test categories saturate at around 50% after 1000 epochs:
![vis_similarity_lr0 001_b100_h2_04021311_acc52 16_e999](https://user-images.githubusercontent.com/17328133/78471814-9b0e5a00-7766-11ea-80ff-486185f826fb.jpg)

The confusion matrixes for train set, val set, and test set are:
```
Train Matrix:
[ Dissimilarity       :  525795   95205	(621000 in all.)]
[ Similarity          :    1728   67272	( 69000 in all.)]

Dissimilarity       :	PR: 99.7%,	RR: 84.7%,	 F1: 91.6%.
Similarity          :	PR: 41.4%,	RR: 97.5%,	 F1: 58.1%.

Val Matrix:
[ Dissimilarity       :  207682   26318	(234000 in all.)]
[ Similarity          :    5590   20410	( 26000 in all.)]

Dissimilarity       :	PR: 97.4%,	RR: 88.8%,	 F1: 92.9%.
Similarity          :	PR: 43.7%,	RR: 78.5%,	 F1: 56.1%.

Test Matrix:
[ Dissimilarity       :   80454    9546	( 90000 in all.)]
[ Similarity          :    3535    6465	( 10000 in all.)]

Dissimilarity       :	PR: 95.8%,	RR: 89.4%,	 F1: 92.5%.
Similarity          :	PR: 40.4%,	RR: 64.6%,	 F1: 49.7%.
```

Therefore, it may need a more powerful model to fit the similarity on large scale datasets.
And it would be helpful if you share your SimilarityNet for ImageNet or some tricks to obtain a good SimilarityNet on large scale dataset.

Thank you.

**Comment by yenchanghsu** [comment]
The training of SimilarityNet converges as fast as a regular classification task. It should converge in 90 epochs like a regular Imagenet classification task. Did you try a larger learning rate?

**Comment by yenchanghsu** [comment]
Please feel free to let me know if any questions.
